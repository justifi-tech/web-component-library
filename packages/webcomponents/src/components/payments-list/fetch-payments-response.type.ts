interface PageInfo {
  has_previous: boolean;
  has_next: boolean;
  start_cursor: string;
  end_cursor: string;
}

interface Card {
  id: string;
  acct_last_four: string;
  brand: string;
  name: string;
  token: string;
  metadata: null;
  created_at: string;
  updated_at: string;
}

interface BankAccount {
  id: string;
  acct_last_four: string;
  name: string;
  brand: string;
  token: string;
  created_at: string;
  updated_at: string;
}

interface PaymentMethod {
  card?: Card;
  bank_account?: BankAccount;
  customer_id: string;
  signature: string;
}

interface ApplicationFee {
  id: string;
  amount: number;
  currency: string;
  created_at: string;
  updated_at: string;
}

interface Refund {
  id: string;
  payment_id: string;
  amount: number;
  description: string | null;
  reason: string | null;
  status: string;
  metadata: null;
  created_at: string;
  updated_at: string;
}

interface DataItem {
  id: string;
  account_id: string;
  amount_disputed: number;
  amount_refunded: number;
  amount_returned: number;
  amount: number;
  amount_refundable: number;
  application_fee_rate_id: string | null;
  balance: number;
  capture_strategy: string;
  captured: boolean;
  created_at: string;
  currency: string;
  description: string;
  statement_descriptor: string;
  disputed: boolean;
  error_code: string | null;
  error_description: string | null;
  fee_amount: number;
  financial_transaction_id: string;
  is_test: boolean;
  metadata: null;
  payment_intent_id: string | null;
  refunded: boolean;
  returned: boolean;
  status: string;
  updated_at: string;
  payment_method: PaymentMethod;
  application_fee: ApplicationFee;
  refunds: Refund[];
  disputes: any[];
  transaction_hold: null;
}

export interface FetchPaymentsResponseType {
  id: string | null;
  type: string;
  page_info: PageInfo;
  data: DataItem[];
}
