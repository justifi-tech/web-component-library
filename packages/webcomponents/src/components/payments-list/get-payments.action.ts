import { Payment } from '../../api';

export const getPayments = async ({
  accountId,
  authToken,
  params,
  onSuccess,
  onError,
  service,
}) => {
  if (!accountId || !authToken) {
    onError('Cannot fetch data without an AccountID and an AuthToken');
    return;
  }

  try {
    const response = await service.fetchPayments(accountId, authToken, params);
    if (!response.error || !response) {
      const pagingInfo = {
        ...response.page_info,
      };

      const payments =
        response.data?.map((dataItem) => new Payment(dataItem)) || [];

      onSuccess({ payments, pagingInfo });
    } else {
      const responseError =
        typeof response.error === 'string'
          ? response.error
          : response.error.message;

      return onError(responseError);
    }
  } catch (error) {
    return onError(error.message || error);
  }
};
