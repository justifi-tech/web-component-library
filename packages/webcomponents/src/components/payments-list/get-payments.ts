import { Payment } from '../../api';

export const makeGetPayments =
  ({ id, authToken, service }) =>
  async ({ params, onSuccess, onError }) => {
    try {
      const response = await service.fetchPayments(id, authToken, params);

      if (!response.error) {
        const pagingInfo = {
          ...response.page_info,
        };

        const payments =
          response.data?.map((dataItem) => new Payment(dataItem)) || [];

        onSuccess({ payments, pagingInfo });
      } else {
        const responseError =
          typeof response.error === 'string'
            ? response.error
            : response.error.message;

        return onError(responseError);
      }
    } catch (error) {
      return onError(error.message || error);
    }
  };
