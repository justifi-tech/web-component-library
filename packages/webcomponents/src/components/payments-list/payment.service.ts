import { Api, IApiResponseCollection, IPayment } from '../../api';
import { config } from '../../../config';

export interface IPaymentService {
  fetchPayments(
    accountId: string,
    authToken: string,
    params: any
  ): Promise<IApiResponseCollection<IPayment[]>>;
}

export class PaymentService implements IPaymentService {
  async fetchPayments(
    accountId: string,
    authToken: string,
    params: any
  ): Promise<IApiResponseCollection<IPayment[]>> {
    const api = Api(authToken, config.proxyApiOrigin);
    const endpoint = `account/${accountId}/payments`;
    return api.get(endpoint, params);
  }
}
