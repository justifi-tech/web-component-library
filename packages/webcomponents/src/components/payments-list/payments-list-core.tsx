import { Component, h, Prop, State, Watch, Event, EventEmitter } from '@stencil/core';
import { PagingInfo, Payment, PaymentsParams, pagingDefaults } from '../../api';
import { MapPaymentStatusToBadge, formatCurrency, formatDate, formatTime } from '../../utils/utils';
import { ComponentError } from '../../api/ComponentError';
import { tableExportedParts } from '../table/exported-parts';
import { StyledHost, TableEmptyState, TableErrorState, TableLoadingState } from '../../ui-components';
import { onFilterChange } from '../../ui-components/filters/utils';

@Component({
  tag: 'payments-list-core'
})
export class PaymentsListCore {
  @State() payments: Payment[] = [];
  @State() loading: boolean = true;
  @State() errorMessage: string;
  @State() paging: PagingInfo = pagingDefaults;
  @State() params: PaymentsParams = {};
  
  @Prop() getPayments: Function;

  @Watch('params')
  @Watch('getPayments')
  updateOnPropChange() {
    this.fetchData();
  }

  @Event({
    eventName: 'payment-row-clicked',
    bubbles: true,
  }) rowClicked: EventEmitter<Payment>;

  @Event({ eventName: 'error-event' }) errorEvent: EventEmitter<ComponentError>;

  componentWillLoad() {
    if (this.getPayments) {
      this.fetchData();
    }
  }

  fetchData(): void {
    this.loading = true;

    this.getPayments({
      params: this.params,
      onSuccess: ({ payments, pagingInfo }) => {
        this.payments = payments;
        this.paging = pagingInfo;
        this.loading = false;
      },
      onError: ({ error, code, severity }) => {
        this.errorMessage = error;
        this.errorEvent.emit({
          errorCode: code,
          message: error,
          severity,
        });
        this.loading = false;
      },
    });
  }

  handleClickPrevious = (beforeCursor: string) => {
    const newParams: any = { ...this.params };
    delete newParams.after_cursor;
    this.params = { ...newParams, before_cursor: beforeCursor };
  };

  handleClickNext = (afterCursor: string) => {
    const newParams: any = { ...this.params };
    delete newParams.before_cursor;
    this.params = { ...newParams, after_cursor: afterCursor };
  };

  rowClickHandler = (e) => {
    const clickedPaymentID = e.target.closest('tr').dataset.rowEntityId;
    if (!clickedPaymentID) return;
    this.rowClicked.emit(this.payments.find((payment) => payment.id === clickedPaymentID));
  };

  setParamsOnChange = (name: string, value: string) => {
    let newParams = { [name]: value };
    this.params = onFilterChange(newParams, this.params);
  }

  clearParams = () => {
    this.params = {};
  }

  get entityId() {
    return this.payments.map((payment) => payment.id);
  }

  get columnData() {
    return [
      ['Date', 'The date and time each payment was made'],
      ['Amount', 'The dollar amount of each payment'],
      ['Status', 'The current status of each payment'],
      ['Type', 'The type of each payment'],
      ['Description', 'The payment description, if you provided one'],
      ['Account Holder', 'The name associated with the payment method'],
      ['Payment Method', 'The brand and last 4 digits of the payment method'],
    ];
  }

  get rowData() {
    return this.payments.map((payment) => [
      {
        type: 'head',
        value: `
          <div class="fw-bold">${formatDate(payment.created_at)}</div>
          <div class="fw-bold">${formatTime(payment.created_at)}</div>
        `,
      },
      formatCurrency(payment.amount),
      {
        type: 'inner',
        value: MapPaymentStatusToBadge(payment.status),
      },
      'payment type!',
      payment.description,
      payment.payment_method.payersName,
      payment.payment_method.lastFourDigits,
    ]);
  }

  get showEmptyState() {
    return !this.loading && !this.errorMessage && this.rowData.length < 1;
  }

  get showErrorState() {
    return !this.loading && !!this.errorMessage;
  }

  get showRowData() {
    return !this.showEmptyState && !this.showErrorState;
  }

  render() {
    return (
      <StyledHost exportparts={tableExportedParts}>
        <payments-list-filters 
          params={this.params} 
          setParamsOnChange={this.setParamsOnChange}
          clearParams={this.clearParams}
        />
        <div class="table-wrapper">
          <table class="table table-hover">
            <thead class="table-head sticky-top" part="table-head">
              <tr class="table-light text-nowrap" part="table-head-row">
                {this.columnData?.map((column) => (
                  <th part="table-head-cell" scope="col" title={Array.isArray(column) ? column[1] : ""}>
                    {!Array.isArray(column) ? column : column[0]}
                  </th>
                ))}
              </tr>
            </thead>
            <tbody class="table-body" part="table-body">
              <TableLoadingState
                columnSpan={this.columnData.length}
                isLoading={this.loading}
              />
              <TableEmptyState
                isEmpty={this.showEmptyState}
                columnSpan={this.columnData.length}
              />
              <TableErrorState
                columnSpan={this.columnData.length}
                errorMessage={this.errorMessage}
              />
              {this.showRowData &&
                this.rowData.map((data, index) => (
                  <tr
                    data-test-id="table-row"
                    data-row-entity-id={this.entityId[index]}
                    onClick={this.rowClickHandler}
                    part={`table-row ${index % 2 ? "table-row-even" : "table-row-odd"}`}
                  >
                    {data.map((dataEntry: any) => {
                      let nestedHtml = dataEntry?.type;
                      if (nestedHtml) {
                        return <td part="table-cell" innerHTML={dataEntry.value}></td>;
                      } else {
                        return <td part="table-cell">{dataEntry}</td>;
                      }
                    })}
                  </tr>
                ))}
            </tbody>
            {this.paging && (
              <tfoot class="sticky-bottom">
                <tr class="table-light align-middle">
                  <td part="pagination-bar" colSpan={this.columnData?.length}>
                    <pagination-menu
                      paging={{
                        ...this.paging,
                        handleClickPrevious: this.handleClickPrevious,
                        handleClickNext: this.handleClickNext,
                      }}
                      params={this.params}
                    />
                  </td>
                </tr>
              </tfoot>
            )}
          </table>
        </div>
      </StyledHost>
    );
  }
}
