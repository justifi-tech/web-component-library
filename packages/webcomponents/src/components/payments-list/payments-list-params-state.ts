import { createStore } from '@stencil/store';
import { PaymentsQueryParams } from '../../api';

const filterParamsInitialState = {};
const propsParamsInitialState = {};

const filterParamsStore = createStore<PaymentsQueryParams>(() => filterParamsInitialState);
const { state: filterParams, on: onQueryParamsChange } = filterParamsStore;

const propsParamsStore = createStore<PaymentsQueryParams>(() => propsParamsInitialState);
const { state: propsParams } = propsParamsStore;

const getRequestParams = (): PaymentsQueryParams => {
  return {
    ...filterParams,
    ...propsParams
  };
};

const clearParams = () => {
  filterParamsStore.reset();
}

export {
  filterParams,
  propsParams,
  onQueryParamsChange,
  clearParams,
  getRequestParams
};
