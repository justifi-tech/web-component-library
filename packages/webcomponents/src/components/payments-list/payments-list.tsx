import { Component, Event, EventEmitter, h, Prop, State, Watch } from '@stencil/core';
import { PaymentService } from '../../api/services/payment.service';
import { makeGetPayments } from './get-payments';
import { ErrorState } from '../details/utils';
import { ComponentError, ComponentErrorCodes, ComponentErrorSeverity } from '../../api/ComponentError';
import JustifiAnalytics from '../../api/Analytics';
import { checkPkgVersion } from '../../utils/check-pkg-version';
import { config } from '../../../config';

/**
  * @exportedPart label: Label for inputs
  * @exportedPart input: The input fields
  * @exportedPart input-invalid: Invalid state for inputs  
  * @exportedPart table-head: Table head
  * @exportedPart table-head-row: Head row
  * @exportedPart table-head-cell: Individual head cell
  * @exportedPart table-body: Body of the table
  * @exportedPart table-row: Row of the table
  * @exportedPart table-cell: Individual cell of the table
  * @exportedPart loading-state-cell: Row for loading state
  * @exportedPart loading-state-spinner: Spinner element for loading state
  * @exportedPart error-state: Row for Error state
  * @exportedPart empty-state: Row for Emtpy state
  * @exportedPart pagination-bar: Pagination bar
  * @exportedPart arrow: Both paging buttons
  * @exportedPart arrow-left: Previous page button
  * @exportedPart arrow-right: Next page button
  * @exportedPart button-disabled: Disabled state for paging buttons
  * @exportedPart previous-button-text: Text for Previous button
  * @exportedPart next-button-text: Text for Next button
*/
@Component({
  tag: 'justifi-payments-list',
  shadow: true
})

export class PaymentsList {
  @State() getPayments: Function;
  @State() errorMessage: string = null;
  
  @Prop() accountId: string;
  @Prop() authToken: string;
  @Prop() apiOrigin?: string = config.proxyApiOrigin;
  
  @Event({ eventName: 'error-event' }) errorEvent: EventEmitter<ComponentError>;

  analytics: JustifiAnalytics;

  componentWillLoad() {
    checkPkgVersion();
    this.analytics = new JustifiAnalytics(this);
    this.initializeGetPayments();
  }

  disconnectedCallback() {
    this.analytics.cleanup();
  }

  @Watch('accountId')
  @Watch('authToken')
  propChanged() {
    this.initializeGetPayments();
  }

  private initializeGetPayments() {
    if (this.accountId && this.authToken) {
      this.getPayments = makeGetPayments({
        id: this.accountId,
        authToken: this.authToken,
        service: new PaymentService(),
        apiOrigin: this.apiOrigin,
      });
    } else {
      this.errorMessage = 'Account ID and Auth Token are required';
      this.errorEvent.emit({
        errorCode: ComponentErrorCodes.MISSING_PROPS,
        message: this.errorMessage,
        severity: ComponentErrorSeverity.ERROR,
      });
    }
  }

  handleErrorEvent = (event) => {
    this.errorMessage = event.detail.message;
    this.errorEvent.emit(event.detail);
  };

  render() {
    if (this.errorMessage) {
      return ErrorState(this.errorMessage);
    }
    return (
      <payments-list-core
        getPayments={this.getPayments}
        onError-event={this.handleErrorEvent}
      />
    );
  }
}
