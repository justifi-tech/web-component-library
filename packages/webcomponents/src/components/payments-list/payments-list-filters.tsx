import { Component, h, Prop } from '@stencil/core';
import { debounce } from 'lodash';
import { filterParams, propsParams, clearParams } from './payments-list-params-state';
import { StyledHost } from '../../ui-components';
import { convertToLocal, convertToUTC } from '../../utils/utils';

@Component({
  tag: 'justifi-payments-list-filters',
  shadow: true
})
export class PaymentsListFilters {
  @Prop() paymentId?: string;
  @Prop() terminalId?: string;
  @Prop() paymentStatus?: string;
  @Prop() createdAfter?: string;
  @Prop() createdBefore?: string;
  
  private debouncedSetParamsOnChange: (name: string, value: string) => void;

  componentWillLoad() {
    // debounced input handler for text input
    this.debouncedSetParamsOnChange = debounce(this.setParamsOnChange, 300);

    const propsToSet = {
      payment_id: this.paymentId,
      terminal_id: this.terminalId,
      payment_status: this.paymentStatus,
      created_after: this.createdAfter,
      created_before: this.createdBefore
    };

    Object.entries(propsToSet).forEach(([key, value]) => {
      if (value) {
        propsParams[key] = value;
      }
    });
  }

  setParamsOnChange = (name: string, value: string) => {
    filterParams[name] = value;
  };

  handleDateInput = (name: string, value: string) => {
    const utcDate = convertToUTC(value, { setExactTime: true });
    this.setParamsOnChange(name, utcDate);
  }

  get paymentStatusOptions() {
    return [
      { label: 'All', value: '' },
      { label: 'Pending', value: 'pending' },
      { label: 'Authorized', value: 'authorized' },
      { label: 'Succeeded', value: 'succeeded' },
      { label: 'Failed', value: 'failed' },
      { label: 'Disputed', value: 'disputed' },
      { label: 'Refunded', value: 'refunded' }
    ]
  }

  render() {
    const filterMenuParams = { ...filterParams}

    return (
      <StyledHost>
        <table-filters-menu params={filterMenuParams} clearParams={clearParams}>
          <div class="grid-cols-2 gap-3 p-1">
            <div class="p-2">
              <form-control-text
                name="payment_id"
                label="Payment ID"
                inputHandler={this.debouncedSetParamsOnChange}
                defaultValue={this.paymentId || filterParams.payment_id}
                disabled={!!this.paymentId}
              />
            </div>
            <div class="p-2">
              <form-control-text
                name="terminal_id"
                label="Terminal ID"
                inputHandler={this.debouncedSetParamsOnChange}
                defaultValue={this.terminalId || filterParams.terminal_id}
                disabled={!!this.terminalId}
              />
            </div>
            <div class="p-2">
              <form-control-select
                name="payment_status"
                label="Status"
                options={this.paymentStatusOptions}
                inputHandler={this.setParamsOnChange}
                defaultValue={this.paymentStatus || filterParams.payment_status}
                disabled={!!this.paymentStatus}
              />
            </div>
            <div class="p-2">
              <form-control-date
                name="created_after"
                label="Created After"
                inputHandler={this.handleDateInput}
                defaultValue={ 
                  convertToLocal(this.createdAfter, { showInputDateTime: true }) ||
                  convertToLocal(filterParams.created_after, { showInputDateTime: true })
                }
                showTime
                disabled={!!this.createdAfter}
              />
            </div>
            <div class="p-2">
              <form-control-date
                name="created_before"
                label="Created Before"
                inputHandler={this.handleDateInput}
                defaultValue={
                  convertToLocal(this.createdAfter, { showInputDateTime: true }) ||
                  convertToLocal(filterParams.created_before, { showInputDateTime: true })
                }
                showTime
                disabled={!!this.createdBefore}
              />
            </div>
          </div>
        </table-filters-menu>
      </StyledHost>
    )
  }
}
