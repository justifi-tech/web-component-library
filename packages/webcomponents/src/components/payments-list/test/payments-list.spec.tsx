import { h } from "@stencil/core";
import { newSpecPage } from "@stencil/core/testing";
import { PaymentsList } from "../payments-list";
import { PaymentsListCore } from "../payments-list-core";
import { Table } from "../../table/table";
import { PaymentService } from "../../../api/services/payment.service";

describe('payments-list', () => {
  it('renders an error message when accountId and authToken are not provided', async () => {
    const page = await newSpecPage({
      components: [PaymentsList, PaymentsListCore],
      html: '<justifi-payments-list></justifi-payments-list>',
    });
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });

  it('renders an error message when accountId is not provided', async () => {
    const page = await newSpecPage({
      components: [PaymentsList, PaymentsListCore],
      html: '<justifi-payments-list auth-token="abc"></justifi-payments-list>',
    });
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });

  it('renders an error message when authToken is not provided', async () => {
    const page = await newSpecPage({
      components: [PaymentsList, PaymentsListCore],
      html: '<justifi-payments-list account-id="abc"></justifi-payments-list>',
    });
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });

  it('emit tokenExpired event when the authToken has expired', async () => {
    const fetchPaymentsMock = jest.spyOn(PaymentService.prototype, 'fetchPayments' as keyof PaymentService);

    fetchPaymentsMock.mockResolvedValue({
      "error": {
        "code": "not_authenticated",
        "message": "Not Authenticated"
      }
    });

    const spy = jest.fn();

    const page = await newSpecPage({
      components: [PaymentsList, PaymentsListCore, Table],
      template: () => <justifi-payments-list account-id="abc" auth-token="abc" onTokenExpired={spy} />,
    });

    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();

    expect(spy).toHaveBeenCalled();
  });
});
