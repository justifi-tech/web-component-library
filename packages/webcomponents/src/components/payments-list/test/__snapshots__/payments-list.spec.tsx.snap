// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`payments-list emit tokenExpired event when the authToken has expired 1`] = `
<justifi-payments-list account-id="abc" auth-token="abc">
  <mock:shadow-root>
    <payments-list-core>
      <justifi-table error-message="Not Authenticated" exportparts="
table-head,table-head-row,table-head-cell,table-body,table-row,table-row-even,
table-row-odd,table-cell,loading-state-cell,loading-state-spinner,error-state,
empty-state,pagination-bar,page-arrow,page-button,page-button-disabled,page-button-text
">
        <mock:shadow-root>
          <div class="table-wrapper">
            <table class="table table-hover">
              <thead class="sticky-top table-head" part="table-head">
                <tr class="table-light text-nowrap" part="table-head-row">
                  <th part="table-head-cell" scope="col" title="The date and time each payment was made">
                    Made On
                  </th>
                  <th part="table-head-cell" scope="col" title="The dollar amount of each payment">
                    Amount
                  </th>
                  <th part="table-head-cell" scope="col" title="The payment description, if you provided one">
                    Description
                  </th>
                  <th part="table-head-cell" scope="col" title="The name associated with the payment method">
                    Cardholder
                  </th>
                  <th part="table-head-cell" scope="col" title="The brand and last 4 digits of the payment method">
                    Payment Method
                  </th>
                  <th part="table-head-cell" scope="col" title="The current status of each payment">
                    Status
                  </th>
                  <th part="table-head-cell" scope="col" title="The unique identifier of each payment">
                    Payment ID
                  </th>
                </tr>
              </thead>
              <tbody class="table-body" part="table-body">
                <tr>
                  <td class="error-state" colspan="7" data-test-id="table-error-state" part="error-state" style="text-align: center;">
                    An unexpected error occurred: Not Authenticated
                  </td>
                </tr>
              </tbody>
              <tfoot class="sticky-bottom">
                <tr class="align-middle table-light">
                  <td colspan="7" part="pagination-bar">
                    <pagination-menu></pagination-menu>
                  </td>
                </tr>
              </tfoot>
            </table>
          </div>
        </mock:shadow-root>
      </justifi-table>
    </payments-list-core>
  </mock:shadow-root>
</justifi-payments-list>
`;

exports[`payments-list renders an error message when accountId and authToken are not provided 1`] = `
<justifi-payments-list>
  <mock:shadow-root>
    <main class="d-flex justify-content-center p-4 text-center" part="detail-empty-state" style="font-size: 1.2rem;">
      <div>
        Account ID and Auth Token are required
      </div>
    </main>
  </mock:shadow-root>
</justifi-payments-list>
`;

exports[`payments-list renders an error message when accountId is not provided 1`] = `
<justifi-payments-list auth-token="abc">
  <mock:shadow-root>
    <main class="d-flex justify-content-center p-4 text-center" part="detail-empty-state" style="font-size: 1.2rem;">
      <div>
        Account ID and Auth Token are required
      </div>
    </main>
  </mock:shadow-root>
</justifi-payments-list>
`;

exports[`payments-list renders an error message when authToken is not provided 1`] = `
<justifi-payments-list account-id="abc">
  <mock:shadow-root>
    <main class="d-flex justify-content-center p-4 text-center" part="detail-empty-state" style="font-size: 1.2rem;">
      <div>
        Account ID and Auth Token are required
      </div>
    </main>
  </mock:shadow-root>
</justifi-payments-list>
`;
